src/test/java/com/sap/sgs/phosphor/fosstars/tool/github/GitHubDataCacheTest.java

```diff
@@ -0,0 +1,56 @@
+package com.sap.sgs.phosphor.fosstars.tool.github;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import org.junit.Test;
+
+public class GitHubDataCacheTest {
+
+  @Test
+  public void testPut() throws IOException {
+    GitHubDataCache<String> gitHubDataCache = new GitHubDataCache<String>();
+    
+    String test = "test1";
+    GitHubProject project = new GitHubProject(new GitHubOrganization(test), test);
+    gitHubDataCache.put(project, test);
+    assertTrue(gitHubDataCache.size() == 1);
+    assertTrue(gitHubDataCache.get(project).get().equals(test));
+
+    GitHubProject project2 = new GitHubProject(new GitHubOrganization(test), test);
+    test = "test2";
+    gitHubDataCache.put(project2, test);
+    assertTrue(gitHubDataCache.size() == 1);
+    assertTrue(gitHubDataCache.get(project2).get().equals(test));
+
+    test = "test3";
+    GitHubProject project3 = new GitHubProject(new GitHubOrganization(test), test);
+    gitHubDataCache.put(project3, test);
+    assertTrue(gitHubDataCache.size() == 2);
+    assertTrue(gitHubDataCache.get(project3).get().equals(test));
+
+    test = "test4";
```

Comment:
Okay, here we know that `put()` works. The rest of checks looks like dups.

Reply:
The remaining line checks for max capacity

Reply:
Then, I think it should use a loop.